File: mobile/src/main/java/com/example/harbit/data/remote/dto/SensorBatchDto.kt
package com.example.harbit.data.remote.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class SensorBatchDto(
    @SerialName("id")
    val id: String,

    @SerialName("deviceId")
    val deviceId: String,

    @SerialName("timestamp")
    val timestamp: Long,

    @SerialName("sampleCount")
    val sampleCount: Int,

    @SerialName("readings")
    val readings: List<SensorReadingDto>
)

File: backend/app/model/dto/sensorBatchDto.py
from pydantic import BaseModel, Field
from model.dto.sensorReadingDto import sensorReadingDto

class sensorBatchDto(BaseModel):
    id: str = Field(..., description="The unique identifier of the sensor batch")
    deviceId: str = Field(..., description="The unique identifier of the device")
    timestamp: int = Field(..., description="The timestamp when the sensor batch was recorded (e.g., 13232642723975)")
    sampleCount: int = Field(..., description="The number of samples in the sensor batch")
    readings: list[sensorReadingDto] = Field(..., description="List of sensor readings in the batch")
